### PR TITLE
Add transaction reminder and more frequent reminder triggers

### PR DESCRIPTION
**Pull request by kberzinch**
- Add reminders for dues transactions outside of dues season
- Trigger dues and travel reminders on login and attendance
- Clean up some method signatures and docblocks

Transaction reminder render with only one purchasable package:
>Hi Kristaps,
>
>You still need to pay dues for Full Year (2022-2023) - Non-Student. Please visit https://apiary-test.robojackets.org to begin the process.
>
>Dues are $200 and may be paid online with a credit or debit card for an additional $6.28 surcharge.
>
>If you would prefer to pay by cash or check, please bring it to an officer or project manager at the shop. Write checks to Georgia Tech, with RoboJackets on the memo line. Don't forget to sign it!
>
>If you have any questions, just reply to this email.
>
>----
>
>To stop receiving emails from *** Apiary Test ***, visit https://subscriptions.pstmrk.it/demo/unsubscribe.

Transaction reminder render with more than one purchasable package:
>Hi Kristaps,
>
>You still need to pay dues for this semester. Please visit https://apiary-test.robojackets.org to begin the process.
>
>Full Year (2022-2023) dues are $100 and Spring 2023 dues are $55. You may pay online with a credit or debit card. Note that we add an additional surcharge for online payments.
>
>If you would prefer to pay by cash or check, please bring it to an officer or project manager at the shop. Write checks to Georgia Tech, with RoboJackets on the memo line. Don't forget to sign it!
>
>If you have any questions, just reply to this email.
>
>----
>
>To stop receiving emails from *** Apiary Test ***, visit https://subscriptions.pstmrk.it/demo/unsubscribe.

### CODE REVIEW

File: resources/views/mail/dues/transactionreminder.blade.php [hunk 1]
```diff
@@ -0,0 +1,21 @@
+Hi {{ $user->preferred_first_name }},
+
+@if(\App\Models\DuesPackage::userCanPurchase($user)->count() > 1)
+You still need to pay dues for this semester. Please visit {{ route('home') }} to begin the process.
+
+@php
+print(implode(" and ", \App\Models\DuesPackage::userCanPurchase($user)->orderByDesc('cost')->get()->map(static fn (\App\Models\DuesPackage $package, int $key): string => $package->name.' dues are $'.intval($package->cost) )->toArray()))
```
Comment (evan10s) [inline resources/views/mail/dues/transactionreminder.blade.php:7]: What happens if there are more than 2 available dues packages? Why not just limit to the top 2?

**Review by evan10s — DISMISSED** [review]

File: resources/views/mail/dues/transactionreminder.blade.php (hunk 1 above)
Comment (kberzinch) [reply resources/views/mail/dues/transactionreminder.blade.php:7]: If there are more than two then you end up with a slightly weirdly phrased list. It still works fine.

In practice there should only be 0, 1, or 2 packages available for any individual person to purchase, so I don't think it should be an issue.

File: app/Jobs/SendReminders.php [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+            return;
+        }
+
+        // check if dues payment reminder should be sent
+        if (! $this->user->is_active &&
+            DuesPackage::userCanPurchase($this->user)->exists() &&
+            $this->user->duesTransactions()->pending()->exists()
+        ) {
+            SendDuesPaymentReminder::dispatch($this->user, 24);
```
Comment (JoshOldenburg) [inline app/Jobs/SendReminders.php:76]: Should this have higher priority than travel? If dues aren't paid, they can't go on a trip.

File: app/Jobs/SendTravelAssignmentReminder.php [hunk 3]
```diff
@@ -25,16 +25,10 @@ class SendTravelAssignmentReminder implements ShouldQueue, ShouldBeUnique
...
      * Create a new job instance.
      */
-    public function __construct(private readonly TravelAssignment $assignment)
+    public function __construct(private readonly TravelAssignment $assignment, int $delay = 48)
     {
         $this->queue = 'email';
-        $this->delay = now()->addHours(48)->hour(10)->startOfHour()->addMinutes(random_int(10, 50));
-
-        if ($this->delay->dayOfWeek === 5) {
-            // do not send reminders on thursdays to reduce the chance of user
```
Comment (JoshOldenburg) [inline app/Jobs/SendTravelAssignmentReminder.php:34]: Is this no longer relevant?

Comment (kberzinch) [reply app/Jobs/SendTravelAssignmentReminder.php:34]: It doesn't seem to work, I don't want to troubleshoot it, and there isn't actually that much downtime (usually)

File: app/Jobs/SendReminders.php (hunk 2 above)
Comment (kberzinch) [reply app/Jobs/SendReminders.php:76]: The travel UX will indicate if dues need to be paid.

**Review by JoshOldenburg — APPROVED** [review]

**Pull request merged**